Apps/APN/components/Dashboard.js

```diff
@@ -0,0 +1,208 @@
+import React, {useState} from 'react'
+import { View, Text, FlatList, StyleSheet, Image, Alert, TouchableOpacity} from 'react-native'
+import CioManager from '../manager/CioManager'
+import ThemedButton from './common/Button'
+import PushNotification from "react-native-push-notification";
+import CioKeyValueStorage from '../manager/KeyValueStorage'
+import { CustomerIO } from 'customerio-reactnative'
+
+const Dashboard = ({navigation}) => {
+
+    const [deviceToken, setDeviceToken] = useState(null)
+    // To get device token
+    PushNotification.configure({
+        onRegister: function (token) {
+          setDeviceToken(token["token"])
+        }
+    });
+    
+    
+    const sendRandomEventTapped = () => {
+        const cioManager = new CioManager()
+        cioManager.randomEvent()
+        alert("Random event tracked successfully.")
+    }
+
+    const settingsTapped = () => {
+        navigation.navigate("SettingsScreen")
+    }
+    const sendCustomEventTapped = () => {
+        navigation.navigate("CustomDataScreen", {
+            featureType : "Custom Event"
+        })
+    }
+
+    const setDeviceAttributesTapped = () => {
+        navigation.navigate("CustomDataScreen", {
+            featureType : "Device Attributes"
+        })
+    }
+
+    const setProfileAttributesTapped = () => {
+        navigation.navigate("CustomDataScreen", {
+            featureType : "Profile Attributes"
+        })
+    }
+
+    const logoutTapped = async() => {
+        const cioManager = new CioManager()
+        cioManager.clearUserIdentity()
+
+        const keyStorageObj = new CioKeyValueStorage()
+        await keyStorageObj.saveLoginStatus(false)
+        navigation.navigate("Login")
+    }
+
+    const renderDashboardButtons = (item) => {
+        return (
+            <ThemedButton onPress={ item.onClick} title={item.key}/>
+        )
+    }
+
+    const showPushPrompt = () => {
+        var options = {"ios" : {"sound" : true, "badge" : true}}
+
+        CustomerIO.showPromptForPushNotifications(options).then(status => {
+            alert("Push permission " + status)
+            console.log("Push permission " + status)
+        }).catch(error => {
+            alert("Could not show prompt.")
+        })
+    }
+
+    const getCurrentPushPermission = () => {
+        CustomerIO.getPushPermissionStatus().then(status => {
+            alert("Push permission status is - " + status)
+            console.log("Push permission status is - " + status)
+          })
+    }
+
+    const registerDeviceToken = () => {
+        if(deviceToken !== null) {
+            CustomerIO.registerDeviceToken(deviceToken)
+            alert("Device token registered successfully")
+            return
+        }
+        alert("Could not register device. Device token unavailable.")
+    }
```

Comment:
Do we still need it? as its not part of the design?

Reply:
I would still like to keep this feature check even though it is not a part of the UI design because it is a feature in react native but not in native SDKs. We should update the apps based on the features that each platform provides without making much changes in the UI for user experience. We would be able to test the SDKs/packages/plugins better if we follow this approach.

Reply:
Feel free to resolve this conversation if this answers your question.